WordPress/Classes/Extensions/UIImageView+Blavatar.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+
+
+/// UIImageView Helper Methods that allow us to download a Blavatar, given a website's "Icon Path"
+///
+extension UIImageView {
+
+    /// Default Settings
+    ///
+    struct BlavatarDefaults {
+
+        /// Default Blavatar's Image Size, in points.
+        ///
+        static let imageSize = 40
+
+        /// Default Blavatar's Image Size, in pixels.
+        ///
+        static var imageSizeInPixels: Int {
+            return imageSize * Int(UIScreen.main.scale)
+        }
+    }
+
+
+    /// Downloads the Blavatar Image, hosted at the specified path. This method will attempt to optimize the URL, so that
+    /// the download Image Size matches `BlavatarDefaults.imageSize`.
+    ///
+    /// TODO: This is a convenience method. Nuke me once we're all swifted.
+    ///
+    /// - Parameter path: Blavatar's url (string encoded) to be downloaded.
+    ///
+    @objc
+    func downloadBlavatar(at path: String) {
+        downloadBlavatar(at: path, placeholderImage: .blavatarPlaceholderImage)
+    }
+
+
+    /// Downloads the Blavatar Image, hosted at the specified path. This method will attempt to optimize the URL, so that
+    /// the download Image Size matches `BlavatarDefaults.imageSize`.
+    ///
+    /// - Parameters:
+    ///     - path: Blavatar's url (string encoded) to be downloaded.
+    ///     - placeholderImage: Yes. It's the "place holder image", Sherlock.
+    ///
+    @objc
+    func downloadBlavatar(at path: String, placeholderImage: UIImage?) {
+        guard let blavatarURL = optimizedURL(for: path) else {
+            image = placeholderImage
+            return
+        }
+
+        setImageWith(blavatarURL, placeholderImage: placeholderImage)
+    }
+
+
+    /// Downloads the Blavatar Image, associated to a given Blog. This method will attempt to optimize the URL, so that
+    /// the download Image Size matches `BlavatarDefaults.imageSize`.
+    ///
+    /// - Parameters:
+    ///     - blog: reference to the source blog
+    ///     - placeholderImage: Yes. It's the "place holder image".
+    ///
+    @objc
+    func downloadBlavatar(for blog: Blog, placeholderImage: UIImage? = .blavatarPlaceholderImage) {
+        guard let blavatarPath = blog.icon, let blavatarURL = optimizedURL(for: blavatarPath) else {
+            image = placeholderImage
+            return
+        }
+
+        guard blog.isHostedAtWPcom && blog.isPrivate() else {
```

Comment:
Using `guard` here feels weird, as if the expected normal scenario was that we're dealing with private sites. I'd turn it into a simpler `if/else`.

Also, `blog.isHostedAtWPcom` is redundant, since only WordPress.com sites can be private:

```objective-c
- (BOOL)isPrivate
{
    return (self.isHostedAtWPcom && [self.settings.privacy isEqualToNumber:@(SiteVisibilityPrivate)]);
}
```